src/backend/base/langflow/components/agentql/agentql_query_doc.py

```diff
@@ -0,0 +1,142 @@
+import json
+from pathlib import Path
+
+import httpx
+from loguru import logger
+
+from langflow.base.data import BaseFileComponent
+from langflow.components.agentql.utils import (
+    AGENTQL_QUERY_DOCUMENTATION,
+    AGENTQL_REST_API_DOCUMENTATION,
+    INVALID_API_KEY_MESSAGE,
+)
+from langflow.io import (
+    DropdownInput,
+    IntInput,
+    MultilineInput,
+    SecretStrInput,
+)
+from langflow.schema import Data
+
+
+class AgentQLQueryDocument(BaseFileComponent):
+    display_name = "Extract Document Data"
+    description = "Extracts structured data from a document using an AgentQL query or a Natural Language description."
+    documentation: str = AGENTQL_REST_API_DOCUMENTATION
+    icon = "AgentQL"
+    name = "AgentQL Query Document"
+
+    VALID_EXTENSIONS = ["jpeg", "png", "pdf", "jpg"]
+
+    SUPPORTED_BUNDLE_EXTENSIONS = []
+
+    inputs = [
+        SecretStrInput(
+            name="api_key",
+            display_name="AgentQL API Key",
+            required=True,
+            password=True,
+            info="Your AgentQL API key. Get one at https://dev.agentql.com.",
+        ),
+        *BaseFileComponent._base_inputs,
+        MultilineInput(
+            name="query",
+            display_name="AgentQL Query",
+            required=False,
+            info=f"The AgentQL query to execute. Read more at {AGENTQL_QUERY_DOCUMENTATION}.",
+            tool_mode=True,
+        ),
+        MultilineInput(
+            name="prompt",
+            display_name="Prompt",
+            required=False,
+            info="A Natural Language description of the data to extract from the page. Alternative to AgentQL query.",
+            tool_mode=True,
+        ),
+        IntInput(
+            name="timeout",
+            display_name="Timeout",
+            info="Timeout in seconds for the request. Increase if data extraction takes too long.",
+            value=900,
+            advanced=True,
+        ),
+        DropdownInput(
+            name="mode",
+            display_name="Request Mode",
+            info="'standard' uses deep data analysis, while 'fast' trades some depth of analysis for speed.",
+            options=["fast", "standard"],
+            value="fast",
+            advanced=True,
+        ),
+    ]
+
+    outputs = [*BaseFileComponent._base_outputs]
+
+    def process_files(self, file_list: list[BaseFileComponent.BaseFile]) -> list[BaseFileComponent.BaseFile]:
+        endpoint = "https://api.agentql.com/v1/query-document-body"
+        headers = {
+            "X-API-Key": self.api_key,
+            "X-TF-Request-Origin": "langflow",
+        }
+
+        if len(file_list) > 1:
+            self.status = "Only one file is supported for AgentQL Query Document."
+            raise ValueError(self.status)
+
+        logger.info(f"Processing file: {file_list[0].path}")
+
+        file = file_list[0]
+
+        if not str(file.path).endswith(tuple(self.VALID_EXTENSIONS)):
```

Comment:
It seems that BaseFileComponent already has a logic to filter out or ignore unsupported files. Can you validate this please?
https://github.com/langflow-ai/langflow/blob/a0c00c015f4a4572cf15e195bb8e6bec37ce72f2/src/backend/base/langflow/base/data/base_file.py#L565